model/dataProvider/providers/ResultsDataProviderService.php

```diff
@@ -0,0 +1,138 @@
+use oat\taoDeliveryRdf\helper\DetectTestAndItemIdentifiersHelper;
+use oat\taoResultServer\models\classes\ResultManagement;
+use oat\taoResultServer\models\classes\ResultServerService;
+use oat\taoSyncClient\model\dataProvider\SyncPackageDataProviderInterface;
+use taoResultServer_models_classes_ReadableResultStorage;
+use taoResultServer_models_classes_WritableResultStorage;
+
+/**
+ * Class ResultDataProviderService
+ * @package oat\taoSyncClient\model\dataProvider\providers
+ */
+class ResultsDataProviderService extends ConfigurableService implements SyncPackageDataProviderInterface
+{
+    /**
+     * @param array $deliveryExecutionIds
+     * @return array
+     * @throws common_exception_NotFound
+     * @throws core_kernel_persistence_Exception
+     */
+    public function getData($deliveryExecutionIds = [])
+    {
+        $results = [];
+        foreach ($deliveryExecutionIds as $deliveryExecutionId) {
+            /** @var DeliveryExecution $deliveryExecution */
+            $deliveryExecution = $this->getServiceProxy()->getDeliveryExecution($deliveryExecutionId);
+            $variables = $this->getDeliveryExecutionVariables($deliveryExecution->getDelivery()->getUri(), $deliveryExecutionId);
+            $results[] = [
+                'deliveryId'          => $deliveryExecution->getDelivery()->getUri(),
+                'deliveryExecutionId' => $deliveryExecutionId,
+                'details'             => $this->getDeliveryExecutionDetails($deliveryExecutionId),
+                'variables'           => $variables,
+            ];
+        }
+        return $results;
+    }
+
+    /**
+     * Get details of a delivery execution
+     *
+     * @param $deliveryExecutionId
+     * @return array
+     */
+    private function getDeliveryExecutionDetails($deliveryExecutionId)
+    {
+        /** @var DeliveryExecution $deliveryExecution */
+        $deliveryExecution = $this->getServiceProxy()->getDeliveryExecution($deliveryExecutionId);
+        try {
+            return [
+                'identifier' => $deliveryExecution->getIdentifier(),
+                'label'      => $deliveryExecution->getLabel(),
+                'test-taker' => $deliveryExecution->getUserIdentifier(),
+                'starttime'  => $deliveryExecution->getStartTime(),
+                'finishtime' => $deliveryExecution->getFinishTime(),
+                'state'      => $deliveryExecution->getState()->getUri(),
+            ];
+        } catch (common_exception_NotFound $e) {
+            return [];
+        }
+    }
+
+    /**
+     * Get variables of a delivery execution
+     *
+     * @param $deliveryId
+     * @param $deliveryExecutionId
+     * @return array
+     * @throws core_kernel_persistence_Exception
+     */
+    private function getDeliveryExecutionVariables($deliveryId, $deliveryExecutionId)
+    {
+        $variables = $this->getResultStorage($deliveryId)->getDeliveryVariables($deliveryExecutionId);
```

Comment:
Cannot test export functionality. It fails on this line with message 

> Uncaught Error: Call to undefined method oat\taoEncryption\Service\Result\EncryptResultService::getDeliveryVariables() in act-tao-client-offline/taoSyncClient/model/dataProvider/providers/ResultsDataProviderService.php:97

Reply:
it's an incorrect implementation of the taoEncryption ResultService. We need to create a bug in the jira for that, not sure that this has to be fixed in this task

Reply:
this PR can be checked without taoEncryption exntension

Reply:
Could you please send me title/description for follow up ticket to fix result storage service in taoEncryption ext.? We have to make it work with encrypted results in the end.